docs/src/main/asciidoc/_admin-guide/subsystem-configuration/MicroProfile_Config_SmallRye.adoc

```diff
@@ -0,0 +1,157 @@
+ the _microprofile-config-smallrye_ subsystem.
+
+[[required-extension]]
+== Required Extension
+
+This extension is included in the standard configurations included in the
+WildFly distribution.
+
+You can also add the extension to a configuration without it either by adding
+an `<extension module="org.wildfly.extension.microprofile.config-smallrye"/>`
+element to the xml or by using the following CLI operation:
+
+[source,ruby]
+----
+[standalone@localhost:9990 /]/extension=org.wildfly.extension.microprofile.config-smallrye:add
+----
+
+== Supported ConfigSources
+
+In addition to the default `ConfigSources` specified by the MicroProfile Config specification
+(environment variables, System properties and `META-INF/microprofile-config.properties` file), the
+`microprofile-config-smallrye` provides additional types of `ConfigSource`
+
+=== ConfigSource from Properties
+
+You can store properties directly in a `config-source` in the subsystem by using the `properties`
+attribute when you add the `config-source`:
+
+[source]
+----
+/subsystem=microprofile-config-smallrye/config-source=props:add(properties={"prop1" = "foo", "prop2" = "bar"})
+----
+
+This results in the XML configuration:
+
+[source,xml]
+----
+<subsystem xmlns="urn:wildfly:microprofile-config-smallrye:1.0">
+    <config-source name="props">
+        <property name="prop1" value="foo"/>
+        <property name="prop2" value="bar"/>
+    </config-source>
+</subsystem>
+----
+
+=== ConfigSource from Directory
+
+You can also read properties from a directory where each file is the name of
+a property and the file content is the value of the property.
+
+For example, let's imagine that the directory `/etc/config/numbers-app/` contains 2 files:
+
+ * the `num.size` file contains the value `5`
+ * the `num.max` file contains the value `100`
+
+We can create a config-source` to access these properties by using the operation:
+
+[source]
+----
+/subsystem=microprofile-config-smallrye/config-source=file-props:add(dir={path=/etc/config/numbers-app})
+----
+
+This results in the XML configuration:
+
+[source,xml]
+----
+<subsystem xmlns="urn:wildfly:microprofile-config-smallrye:1.0">
+    <config-source name="file-props">
+        <dir path="/etc/config/numbers-app"/>
+    </config-source>
+</subsystem>
+----
+
+With that configuration, any application deployed in WildFly can use the `num.size` and
+`num.max` properties that are stored in the directory:
+
+[source,java]
+----
+@Inject
+@ConfigProperty(name = "num.size")
+int numSize; // <1>
+
+@Inject
+@ConfigProperty(name = "num.max")
+int numMax; // <2>
+----
+<1> will be set to 5
+<2> will be set to 100
+
+[NOTE]
+This corresponds to the layout used by https://docs.openshift.com/enterprise/3.2/dev_guide/configmaps.html[OpenShift ConfigMaps].
+The `dir` value corresponds to the `mountPath` in the ConfigMap definition in OpenShift or Kubernetes.
+
```

Comment:
This seems to me a little bit poor description regarding OpenShift integration. So does this mean that only properties configured in ConfigMap will be supported?

Env. variables defined e.g in Pod or DeploymentConfig will be ignored?